compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -255,10 +266,15 @@ static LogicalResult canTargetIntrinsic(const GPUMatmulShapeType &problem,
   // established after we sweep the different tile sizes for a problem config.
   // Once a precise threshold is established, replace 4 with the threshold and
   // remove this todo.
-  if (!mustBeAligned &&
-      (problem.mSizes.back() < 4 || problem.nSizes.back() < 4 ||
-       problem.kSizes.back() < 4)) {
-    return failure();
+  if (llvm::all_equal({problem.mSizes.size(), problem.nSizes.size(),
+                       problem.kSizes.size(), size_t{1}}) &&
+      problem.batchSizes.empty()) {
+    int64_t mSize = problem.mSizes.back();
+    int64_t nSize = problem.nSizes.back();
+    if ((mSize <= kVerySkinnyDimThreshold && (nSize > preferredSubgroupSize)) ||
+        (nSize <= kVerySkinnyDimThreshold && (mSize > preferredSubgroupSize))) {
+      return failure();
+    }
```

Comment:
Does this leave tiny gemms to tile and fuse pipeline? It looks to me that, one condition that won't satisfy above is:
 - mSize <= kVerySkinnyDimThreshold 
 - nSize <= kVerySkinnyDimThreshold 

(Note I'm assuming kVerySkinneyDimThreshold is smaller than subgroup size) In this case it is a tiny gemm but we'll return `success()`. From the context it looks like we don't want to do that.

Reply:
Yes that is correct. Since these GEMMs are not a bottleneck for any model and we dont have specialized pipeline for them I figured this would be okay. Do you think we should add the other check back?

Reply:
Yes, I think we should disable those to run through tile and fuse as that is more conceptually correct.

Also, is this PR a temporary measure to get skinny gemms working? Is the long term plan to get reduction pipeline to support skinny gemms?

Reply:
I think this PR is attempting to support everything that the the reduction pipeline doesn't support today. For example it doesn't support anything with multiple M,N,K dims and any batch dims. I think yes the long term strategy should be to support those through reduction pipeline.

Reply:
Make sense, because theoretically speaking tile and fuse isn't in going to deliver the best perf for those shapes. I think you can ignore my comments now. When we find those situation to be a bottleneck, we can always come back to flip it.

Probably file a ticket to track this? I'm going to mark this as resolved.